include/cudaq/Optimizer/Builder/Factory.h

```diff
@@ -107,6 +107,14 @@ inline mlir::LLVM::LLVMStructType stdVectorImplType(mlir::Type eleTy) {
   return mlir::LLVM::LLVMStructType::getLiteral(ctx, eleTys);
 }
 
+inline mlir::LLVM::LLVMStructType stdStringImplType(mlir::Type eleTy) {
+  auto *ctx = eleTy.getContext();
+  auto elePtrTy = cudaq::opt::factory::getPointerType(eleTy);
+  auto i64Ty = mlir::IntegerType::get(ctx, 64);
+  llvm::SmallVector<mlir::Type> eleTys = {elePtrTy, i64Ty, i64Ty, i64Ty};
```

Comment:
I thought we were going for span types to be `{T*, i64}`?

Reply:
Yes, we are.